validator/sawtooth_validator/execution/py_executor.py

```diff
@@ -0,0 +1,178 @@
+
+        global_state_db = NativeLmdbDatabase(global_state_db_filename)
+
+        component_thread_pool = InstrumentedThreadPoolExecutor(
+            max_workers=10,
+            name='Component')
+
+        component_dispatcher = Dispatcher()
+        component_service = Interconnect(
+            bind_component,
+            component_dispatcher,
+            secured=False,
+            heartbeat=False,
+            max_incoming_connections=20,
+            monitor=True,
+            max_future_callback_workers=10)
+
+        context_manager = ContextManager(global_state_db)
+        state_view_factory = StateViewFactory(global_state_db)
+
+        transaction_executor = TransactionExecutor(
+            service=component_service,
+            context_manager=context_manager,
+            settings_view_factory=SettingsViewFactory(state_view_factory),
+            scheduler_type=scheduler_type,
+            invalid_observers=[])
+
+        component_service.set_check_connections(
+            transaction_executor.check_connections)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_RECEIPT_ADD_DATA_REQUEST,
+            tp_state_handlers.TpReceiptAddDataHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_EVENT_ADD_REQUEST,
+            tp_state_handlers.TpEventAddHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_STATE_DELETE_REQUEST,
+            tp_state_handlers.TpStateDeleteHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_STATE_GET_REQUEST,
+            tp_state_handlers.TpStateGetHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_STATE_SET_REQUEST,
+            tp_state_handlers.TpStateSetHandler(context_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_REGISTER_REQUEST,
+            processor_handlers.ProcessorRegisterHandler(
+                transaction_executor.processor_manager),
+            component_thread_pool)
+
+        component_dispatcher.add_handler(
+            validator_pb2.Message.TP_UNREGISTER_REQUEST,
+            processor_handlers.ProcessorUnRegisterHandler(
+                transaction_executor.processor_manager),
+            component_thread_pool)
+
+        component_dispatcher.start()
+        component_service.start()
+
+        self._executor = transaction_executor
+        self._context_manager = context_manager
+
+        self._component_service = component_service
+        self._component_dispatcher = component_dispatcher
+
+        self._schedulers = {}
```

Comment:
What is your plan for integrating this into the existing validator? Some of these things need to be used by other components in the system and I don't think it makes sense to create them here.

Reply:
My thinking was that we could create a client instance of Interconnect and Dispatcher. I think that would work.

Reply:
You still need a solution for how you bring in objects that you don't create in this constructor. For example, the global_state_db. Maybe you can create Interconnect and Dispatcher here, but that is a departure from the current pattern and requires a user facing change (they must now also configure the client bind endpoint. fwiw I think client and component should have been separate from the beginning).

Reply:
I don't see why global_state_db can't be instantiated inside the Transaction Execution python code. That is the only portion of the code that should write to state. There may be other parts of the code that read from state. I don't see how to pass Interconnect and the dispatcher over the FFI, so having a breaking Api change seems like it is the only possibility. And it kind of makes sense, too.